localstack-core/localstack/services/cloudformation/engine/template_utils.py

```diff
@@ -184,30 +184,50 @@ def resolve_condition(
                     map_name, top_level_key, second_level_key = v
                     if isinstance(map_name, dict) and "Ref" in map_name:
                         ref_name = map_name["Ref"]
-                        param = parameters.get(ref_name)
+                        param = parameters.get(ref_name) or resolve_pseudo_parameter(
+                            account_id, region_name, ref_name, stack_name
+                        )
                         if not param:
                             raise TemplateError(
                                 f"Invalid reference: '{ref_name}' does not exist in parameters: '{parameters}'"
                             )
-                        map_name = param.get("ResolvedValue") or param.get("ParameterValue")
+                        map_name = (
+                            (param.get("ResolvedValue") or param.get("ParameterValue"))
+                            if isinstance(param, dict)
+                            else param
+                        )
```

Comment:
nit: I think it would be clearer here to keep those cases (parameters vs. pseudo-parameters) better separated for clarity. E.g. explicitly handling `resolve_pseudo_parameter` when the value starts with a `AWS::...` instead of using it as a fallback here. (same applies for the cases further below)

Reply:
Thanks for the suggestion. It's implemented int #11693